src/ReactiveUI.Maui/RoutedViewHost.cs

```diff
@@ -178,6 +178,10 @@ protected virtual IObservable<Page> PagesForViewModel(IRoutableViewModel? vm)
         ret.ViewModel = vm;
 
         var pg = (Page)ret;
+        MainThread.BeginInvokeOnMainThread(() =>
+        {
+            pg.Title = vm.UrlPathSegment;
+        });
```

Comment:
@ChrisPulman there is the problem with setting title - setting it directly will break later binding.

Also if there are cross-thread issues may be it would be better to do smth like this:


```
			Router?
				.Navigate
				.ObserveOn(RxApp.MainThreadScheduler) // <-- here
				.SelectMany(_ => PagesForViewModel(Router.GetCurrentViewModel()))
				.SelectMany(async page =>

```
?

Reply:
I believe that it will only affect the ability to bind the Title, which with other platforms such as Xamarin we set the Title property too. 
However there could be some argument that we shouldn't be setting this property on any platform to allow the implementor to choose what and how the properties are set or bound. 
There should be commonality across the platforms in my opinion.
I'll ask @glennawatson to add his opinion on this.

Reply:
@ChrisPulman i can't say about Xamarin, may be there are some difference between Xamarin bindings & MAUI compiled bindings, but the fact is - setting `Title` on navigating breaks bindings. I'v commented this code in my local `RoutedViewHost` implementation, and this saves :)